forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestJWTClaims.java

```diff
@@ -0,0 +1,211 @@
+        try {
+            return Optional.ofNullable(claimsSet.getStringClaim(claimName));
+        } catch (ParseException e) {
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getStringClaims() {}", errorString, e);
+            throw new DynamicClientRegistrationException(errorString, errorType);
+        }
+    }
+
+    private boolean stringClaimValueIsValid(Optional<String> claimValue){
+        return claimValue.isPresent() && !StringUtils.isEmpty(claimValue.get());
+    }
+
+
+    /**
+     * getRequiredStringListClaims Obtains a list of strings containing the values for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName  {@code String} the name of the SSA claim to obtain from the ssaClaims
+     * @return A non null or empty {@code List<String>} containing the ssa claim value
+     * @throws @DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is
+     * empty.
+     */
+    public List<String> getRequiredStringListClaims(String claimName) throws DynamicClientRegistrationException {
+            Optional<List<String>> values = getOptionalStringListClaims(claimName);
+            if (!stringListClaimsAreValid(values)) {
+                String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                        "is required";
+                log.debug("getStringClaims() {}", errorString);
+                throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+            }
+            return values.get();
+
+    }
+
+    private boolean stringListClaimsAreValid(Optional<List<String>> values) {
+        return values.isPresent() && !values.get().isEmpty();
+    }
+
+    public long getRequiredLongValue(String claimName) throws DynamicClientRegistrationException {
+        Optional<Long> value = getOptionalLongValue(claimName);
+        if (value.isEmpty()){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                    "is required";
+            log.debug("getRequiredLongValue() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString, getErrorType(claimName));
+        }
+        return value.get();
+    }
+
+
+    public Optional<Long> getOptionalLongValue(String claimName) throws DynamicClientRegistrationException {
+        try {
+            return Optional.ofNullable(this.claimsSet.getLongClaim(claimName));
+        } catch (ParseException pe){
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getOptionalLongValue() {}", errorString, pe);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.

Reply:
we worked out this should be log.info